### PR TITLE
Make social preview text editable

### PR DESCRIPTION
**Pull request by Will-Howard**
Using designs [here](https://www.figma.com/file/d09iVzNSpZp9GyWGcfkWmX/Q2-2023?node-id=863%3A49255&mode=dev)

Screenshot 2023-07-11 at 11 39 33

The only significant deviation from the designs is that I've added the "use default text" button to go back to using the text from the body of the post. This is necessary because in the case where they backspace all their custom text, it would be weird to immediately go back to using the body text (i.e. I expect a lot of people would delete the existing text before writing anything), so there needs to be a distinction between the field just being empty vs using the default version. Note that in the case where they actually submit with the field completely empty, it will still fall back to using the default version, I don't think we need to explain this to the user though

cc @agnesstenlund 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205002209981928) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/form-components/SocialPreviewUpload.tsx [hunk 1]
```diff
@@ -100,103 +141,224 @@ mdi.use(markdownItSup)
...
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const { description, fallbackImageUrl } = useMemo(() => buildPreviewFromDocument(document), [document.contents?.originalContents, document.contents?.dataWithDiscardedSuggestions])
+  // This handling here is a workaround for a bug in the Input component, you can ignore it
+  // and just assume `value` is the source of truth here
+  //
+  // Details of the bug: The Input component manages its own size with some javascript which sets
+  // its height explicitly. When you type one character at a time this is fine, but when a large
+  // block of text is pasted in it fails to expand the height enough (I think it only does one line
+  // per update) and a scroll is introduced. This workaround makes it so large updates are incrementally
+  // added when the length reaches the danger zone of introducing a scroll.
```
Comment (Will-Howard) [inline packages/lesswrong/components/form-components/SocialPreviewUpload.tsx:222]: Worth reading

File: packages/lesswrong/components/form-components/SocialPreviewUpload.tsx [hunk 2]
```diff
@@ -100,103 +141,224 @@ mdi.use(markdownItSup)
...
 
-const SocialPreviewUpload = ({name, document, updateCurrentValues, clearField, label, croppingAspectRatio, classes}: {
-  name: string,
-  document: PostsEditWithLocalData,
-  updateCurrentValues: UpdateCurrentValues,
-  clearField: Function,
-  label: string,
-  croppingAspectRatio?: number,
-  classes: ClassesType
+const SocialPreviewTextEdit = ({
```
Comment (Will-Howard) [inline packages/lesswrong/components/form-components/SocialPreviewUpload.tsx:206]: I haven't set any hard limit on the max length people can write. I expect common sense + the nudge of the field being kind of small will do the job, and there isn't really a problem with writing a really long paragraph here (some places like slack actually do show a lot of preview text)

File: packages/lesswrong/components/form-components/SocialPreviewUpload.tsx [hunk 3]
```diff
@@ -1,28 +1,34 @@
...
+
+const DESCRIPTION_HEIGHT = 56; // 3 lines
 
 const styles = (theme: ThemeType): JssStyles => ({
   root: {
     display: "grid",
     gridTemplateColumns: "minmax(201px, 1fr) minmax(170px, 269px)",
 
-    [theme.breakpoints.down('xs')]: {
+    [theme.breakpoints.down("xs")]: {
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/form-components/SocialPreviewUpload.tsx:28]: Your auto-formatter has good taste

**Review by oetherington — APPROVED** [review]
Nice! Looks good to me!

**Pull request merged**